### DISCUSSION TITLE
Datas: FindSystemTimeZoneById - Erro com String "Pacific/Auckland"

### DISCUSSION THREAD

**Discussion by Dallonso** [balta-io/2801#55] [Perguntas e Respostas] [answered]
Boa noite André, tudo certo?

Dei uma procurada e não encontrei no fórum algo sobre um problema que me deparei durante o assunto sobre **Time Zone**.
Quando dou um _dotnet run_ no terminal, o programa da um erro, como se não reconhecesse a string _"Pacific/Auckland"_.

Abaixo imagem do erro:
![image](https://user-images.githubusercontent.com/15056126/156277043-a051b280-fd7b-4235-8ee7-13594c5434eb.png)

Me parece que o programa esta tentando buscar essa informação no meu pc e não esta tendo exito, seria isso?

![image](https://user-images.githubusercontent.com/15056126/156277244-f87e7cda-1df6-42eb-980b-27d874852d19.png)

Abaixo meu código, para analise:
```
var utcDate = DateTime.UtcNow;

Console.WriteLine(DateTime.Now);
Console.WriteLine(utcDate);
Console.WriteLine(utcDate.ToLocalTime());

var timezoneAustralia = TimeZoneInfo.FindSystemTimeZoneById("Pacific/Auckland");
Console.WriteLine(timezoneAustralia);

var horaAustralia = TimeZoneInfo.ConvertTimeFromUtc(utcDate,timezoneAustralia);
Console.WriteLine(horaAustralia);
```

Sabe me dizer se é algo que estou fazendo errado, ou esta relacionado ao dotnet 6.0, e ainda, se consigo contornar?

Grato pelo suporte.
Diego.

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @Dallonso , conforme comentei no curso, os valores dos TimeZones podem mudar de acordo com a máquina (SO).
Desta forma recomendo executar primeiro o código para listar os TimeZones e depois escolher um deles para te atender:

```csharp
// var timezones = TimeZoneInfo.GetSystemTimeZones();
// foreach (var timezone in timezones)
// {
//     Console.WriteLine(timezone.Id);
//     Console.WriteLine(timezone);
//     Console.WriteLine(TimeZoneInfo.ConvertTimeFromUtc(dateTime, timezone));
//     Console.WriteLine("_________");
// }

```

**Reply by Dallonso** [reply]
Opa @andrebaltieri 
Maravilha, me desculpe essa questão, acho que não me atentei durante o curso.

Vlw

**Reply by andrebaltieri** [reply]
💜
